src/main/java/io/hops/util/featurestore/ops/write_ops/FeaturestoreInsertIntoTrainingDataset.java

```diff
@@ -89,9 +90,11 @@ public void write()
       throw new IllegalArgumentException("Dataframe to insert cannot be null, specify dataframe with " +
         ".setDataframe(df)");
     }
-    if (mode==null || !mode.equalsIgnoreCase("overwrite"))
+
+    List<String> supportedModes = Arrays.asList(Constants.SPARK_OVERWRITE_MODE, Constants.SPARK_APPEND_MODE);
+    if (mode==null || !supportedModes.stream().anyMatch(x -> x.equalsIgnoreCase(mode)))
```

Comment:
But append only works for the old connector, right? what if user sets `mode="append"` now and `data_format="tfrecords"` ?

Reply:
I believe in Python you added a check for this

Reply:
it will raise `IllegalArgumentException: 'Append mode is not supported'`.